src/common/tracer.cc

```diff
@@ -55,6 +65,37 @@ bool Tracer::is_enabled() const {
   return g_ceph_context->_conf->jaeger_tracing_enable;
 }
 
+void encode(const jspan_context& span_ctx, bufferlist& bl, uint64_t f) {
+  if (span_ctx.IsValid()) {
+    using namespace opentelemetry;
+    using namespace trace;
+    ENCODE_START(1, 1, bl);
+    encode_nohead(std::string_view((char*)span_ctx.trace_id().Id().data(), TraceId::kSize), bl);
+    encode_nohead(std::string_view((char*)span_ctx.span_id().Id().data(), SpanId::kSize), bl);
+    encode(span_ctx.trace_flags().flags(), bl);
+    ENCODE_FINISH(bl);
+  }
+}
+
+void decode(jspan_context& span_ctx, bufferlist::const_iterator& bl) {
+  if (bl.get_remaining() == 0) {
+    return ;
+  }
+  using namespace opentelemetry;
+  using namespace trace;
+  const char* trace_id_ptr;
+  const char* span_id_ptr;
+  uint8_t flags;
+  DECODE_START(1, bl);
+  bl.get_ptr_and_advance(TraceId::kSize, &trace_id_ptr);
+  bl.get_ptr_and_advance(SpanId::kSize, &span_id_ptr);
+  decode(flags, bl);
+  DECODE_FINISH(bl);
+  nostd::span<uint8_t, TraceId::kSize> trace((uint8_t*)trace_id_ptr, TraceId::kSize);
+  nostd::span<uint8_t, SpanId::kSize> span((uint8_t*)span_id_ptr, SpanId::kSize);
+  SpanContext ctx(TraceId(trace), SpanId(span), TraceFlags(flags), true);
```

Comment:
i think we can save one copy:
```suggestion
  SpanContext ctx(
    TraceId(nostd::span<uint8_t, TraceId::kSize>(trace_id_ptr, TraceId::kSize),
    SpanId(nostd::span<uint8_t, SpanId::kSize>(span_id_ptr, SpanId::kSize),
    TraceFlags(flags), 
    true);
```